### PR TITLE
Sink service Fix up

### PR DESCRIPTION
**Pull request by vtereso**
# Changes
Adjust KService to use newest format as per [docs](https://github.com/knative/docs/blob/master/docs/serving/samples/hello-world/helloworld-go/service.yaml)
Update scripts to install Istio since it is no longer bundled in the release artifacts
Loop ` ko` on failures to account for networking timeouts
Remove conflicting PORT as mentioned [here](https://github.com/knative/serving/pull/4050/files)
Hardcoding `INSTALLED_NAMESPACE` env since KService does not implement Downward API
Updated docs

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by vtereso** [comment]
The pod creation logic relevant to the `PORT` variable creation can be seen [here](https://github.com/knative/serving/blob/befc0822427d7c6a712e73538d98444bf37aaeb3/pkg/reconciler/revision/resources/deploy.go#L93)

**Comment by mnuttall** [comment]
Hi @vtereso thank you for this - please could you also update the `install_knative_*` methods used by https://github.com/tektoncd/experimental/blob/master/webhooks-extension/test/install_prereqs.sh to install Knative? I take it that this change requires a certain level of Knative to function correctly?

**Comment by vtereso** [comment]
I believe that ` webhooks-extension/development_install.sh` is an artifact and can be removed, but I made a bunch of doc changes so I want to get an ok before I do that and squash.

**Comment by vtereso** [comment]
Tested everything and works :)

File: webhooks-extension/test/README.md [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
-- `test/install_dashboard_and_extension.sh`
+- Run `install_dashboard_and_extension.sh`
 
+# Testing 
 ## Test with a webhook
+_Note: Your git provider must be able to reach the address of this extension's sink. The sink is deployed with a Knative service, so you may need to configure Knative serving. We recommend [setting up a custom domain](https://knative.dev/v0.3-docs/serving/using-a-custom-domain/) with the extension `.nip.io`. The `install_prereqs.sh` script patches the workstation IP is patched into the knative serving config-map (config-domain) as mentioned in the previous link._
 
-As we state in config.sh, 
-```
-# Note that to receive webhooks, your github must be able to http POST to your Tekton installation.
```
[the remark this answers is not in the document]
Comment (vtereso) [reply webhooks-extension/test/README.md]: Change diff seems weird. https://github.com/tektoncd/experimental/blob/4c65d06dc1c29de18eb083d6f27966a3d8694c1f/webhooks-extension/test/README.md#test-with-a-webhook

**Pull request merged**